### PR TITLE
Fix(edit some UI and some functionality)

### PR DESCRIPTION
**Pull request by mohsinayaz**
1:- Edit some UI changes according to the instructions and add some new things also.
2:- Remove the procedure to pass data form URL. Get user by its id and show the user details on page

### CODE REVIEW

File: src/app/home/home.component.scss [hunk 1]
```diff
@@ -1,29 +1,56 @@
...
+        flex-wrap: wrap;
+        max-width: 100%;
+    
+        .card {
+            align-items: center;
             margin: 1rem;
-            width: 136px;
-            border-radius: 50%;
+            width: 500px;
+            border: 1px solid #00000f;
```
Comment (AhsanAyaz) [inline src/app/home/home.component.scss]: Would have been nice to use constants for scss variables.

File: src/app/home/home.component.scss [hunk 2]
```diff
@@ -1,29 +1,56 @@
...
+            z-index: 5;
+    
+            ::after {
+                content: "";
+                display: block;
+                position: absolute;
+                width: 100%;
+                height: 74px;
+                top: 0;
+                background: #f9f9f9;
```
Comment (AhsanAyaz) [inline src/app/home/home.component.scss]: Same here. Could have used a constant.

File: src/app/user-detail/user-detail.component.ts [hunk 3]
```diff
@@ -1,32 +1,39 @@
...
+    private route: ActivatedRoute,
+    private userService: UsersService
     ) { }
 
   ngOnInit(): void {
-    this.route.queryParams.subscribe(params => {
-      this.user = JSON.parse(params.data);
+    let userId: number;
+    this.route.params.subscribe((params:Params) => {
+      userId = +params['userId'];
```
Comment (AhsanAyaz) [inline src/app/user-detail/user-detail.component.ts]: A good practice is to use `userId = Number(params['userId']);` instead

File: src/app/user-detail/user-detail.component.ts [hunk 4]
```diff
@@ -1,32 +1,39 @@
...
     })
-    this.detail = this.user.name;
-    this.title = 'Hi, My name is';
+    if (!this.userService.newData) {
+      this.userService.getUsers().subscribe(data => {
```
Comment (AhsanAyaz) [inline src/app/user-detail/user-detail.component.ts]: We're not using the `data` variable at all. We can just use `() =>` instead.

**Review by AhsanAyaz — CHANGES_REQUESTED** [review]
Looks good overall 🎉 Great job.
Just a few comments

File: src/app/home/home.component.scss (hunk 1 above)
Comment (mohsinayaz) [reply src/app/home/home.component.scss]: done

File: src/app/home/home.component.scss (hunk 2 above)
Comment (mohsinayaz) [reply src/app/home/home.component.scss]: done

Comment (mohsinayaz) [reply src/app/home/home.component.scss]: done

File: src/app/user-detail/user-detail.component.ts (hunk 3 above)
Comment (mohsinayaz) [reply src/app/user-detail/user-detail.component.ts]: done

File: src/app/user-detail/user-detail.component.ts (hunk 4 above)
Comment (mohsinayaz) [reply src/app/user-detail/user-detail.component.ts]: done

**Pull request merged**